features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,289 @@
+
+## Tutorial
+
+### Name and Version
+
+`present-proof`, version 2.0
+
+### Key Concepts
+
+This protocol is about the messages to support the presentation of verifiable claims, not about the specifics of particular verifiable presentation formats. [DIDComm attachments](../../concepts/0017-attachments/README.md) are deliberately used in messages to make the protocol agnostic to specific verifiable presentation format payloads. Links are provided in the message data element descriptions to details of specific verifiable presentation implementation data structures.
+
+Diagrams in this protocol were made in draw.io. To make changes:
+
+- upload the drawing HTML from this folder to the [draw.io](https://draw.io) site (Import From...GitHub),
+- make changes,
+- export the picture and HTML to your local copy of this repo, and
+- submit a pull request.
+
+### Roles
+
+The roles are `verifier` and `prover`.  The `verifier` requests the presentation of a proof and verifies the presentation, while the `prover` prepares the proof and presents it to the verifier. Optionally, although unlikely from a business sense, the `prover` may initiate an instance of the protocol using the `propose-presentation` message.
+
+### States
+
+#### States for Verifier
+
+* request-sent
+* proposal-received
+* presentation-received
+* abandoned
+* done
+
+#### States for Prover
+
+* request-received
+* proposal-sent
+* presentation-sent
+* abandoned
+* done
+
+For the most part, these states map onto the transitions shown in the choreography diagram in obvious ways. However, a few subtleties are worth highlighting:
+
+* The final transitions for the verifier (Send Presentation Reject, Send Presentation Ack, and Receive Presentation Reject) all result in a final `done` state, but this `done` may or may not be the verifiers intended outcome. We may want to tweak this in a future version of the protocol.
+
+> To Do: Should the previous sentence be addressed? Removed?
+
+* A similar situation exists with the final transitions for the Prover and its final transitions (Send Presentation Reject, Receive Presentation Ack, and Receive Presentation Reject).
+* When a Prover makes a (counter-)proposal, it transitions to the `proposal-sent` state. This state is only present by implication in the choreography diagram; it essentially equates to the null or begin state in that the Prover does nothing until a presentation request arrives, triggering the leftmost transition for the Prover.
+
+Errors might occur in various places. For example, a Verifier might time out waiting for the Prover to supply a presentation. Errors trigger a `problem-report`. In this version of the protocol, all errors cause the state of both parties (the sender and the receiver of the `problem-report`) to revert to `null` (meaning it is no longer engaged in the protocol at all). Future versions of the protocol may allow more granular choices.
```

Comment:
```suggestion
Errors might occur in various places. For example, a Verifier might time out waiting for the Prover to supply a presentation. Errors trigger a `problem-report`. In this version of the protocol, all errors cause the state of both parties (the sender and the receiver of the `problem-report`) transition to `abandoned` (meaning it is no longer engaged in the protocol at all). Future versions of the protocol may allow more granular choices.
```

Reply:
Will make the change.  Can we drop the last sentence since that is true of every word and phrase in the RFC?

Reply:
Fixed in next commit - removed the final sentence and the related "To Do".

Reply:
> Can we drop the last sentence since that is true of every word and phrase in the RFC?

If you're referring to this then sure:

> Future versions of the protocol may allow more granular choices.